packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -133,6 +135,16 @@ class CupertinoSwitch extends StatefulWidget {
   /// Defaults to a slightly transparent [activeColor].
   final Color? focusColor;
 
+  /// The color to use for the accessibility label when the switch is on.
+  ///
+  /// Defaults to [CupertinoColors.white] when null.
+  final Color? onLabelColor;
+
+  /// The color to use for the accessibility label when the switch is off.
+  ///
+  /// Defaults to [Color(0xFFB3B3B3)] when null.
```

Comment:
Why's the default on color a dynamic color while the off color is a hard coded one? Isn't the off color supposed to change in dark mode / high contrast mode etc.?

Reply:
Huh I asked the same question in https://github.com/flutter/flutter/pull/39993 and it seems the answer is no. But does `onLabelColor` respond to dark mode / high contrast?

Reply:
Nice spot, I forgot to check for high contrast.

It seems that `onLabelColor` stays the same in high contrast or dark mode. However, the `offLabelColor` becomes white in high contrast in both light and dark modes.

This behaviour doesn't correspond to any of the `CupertinoColors` though.

Screenshots from Settings app, iOS 16.5.1:

##### Light mode
<img width="462" alt="lightmode" src="https://github.com/flutter/flutter/assets/82336674/5df5362f-7d4a-42ab-821d-4268cf406ac3">

##### Light mode, High contrast
<img width="462" alt="image" src="https://github.com/flutter/flutter/assets/82336674/a3af3c52-7158-40a0-bf28-226b44ded81d">

##### Dark mode
<img width="462" alt="image" src="https://github.com/flutter/flutter/assets/82336674/5e90b0c0-79a9-437a-909d-dc71f59bbe59">

##### Dark mode, High contrast
<img width="462" alt="image" src="https://github.com/flutter/flutter/assets/82336674/99423340-2832-45d1-91c2-a3b99f49dc26">